sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -404,6 +404,9 @@ trait DivModLike extends BinaryArithmetic {
 
   protected def decimalToDataTypeCodeGen(decimalResult: String): String = decimalResult
 
+  // Whether we should check overflow or not in ANSI mode.
+  protected def checkDivideOverflow: Boolean = false
```

Comment:
Is this simply `left.dataType == LongType`? For `Divide`, the child is either double or decimal, so won't hit this branch.

Under the hood, what we really want to handle is `Long.MinValue / -1`, so check `left.dataType == LongType` looks reasonable.

Reply:
No, Remainder's left can be Long as well